ui/app/components/app/modals/account-modal-container/account-modal-container.component.js

```diff
@@ -40,12 +40,13 @@ AccountModalContainer.contextTypes = {
 AccountModalContainer.defaultProps = {
   showBackButton: false,
   children: null,
+  backButtonAction: _ => { /* no-op */ },
```

Comment:
Could this be set to `undefined` instead? I don't think the function is needed.

Reply:
If it's `undefined` and _somehow_ gets called, that'll result in an `Uncaught TypeError`, non?

Reply:
Yes, but it is never called. It's just passed to a `div`. That's why this worked before @whymarrh 's recent refactor.

Reply:
`undefined` seems to work sans warning... idk but it feels weird to me to use it in place of something that is intended to be a function? if that's what we want however, I'm fine with it

Reply:
I think this is best left as a function since that's what its type is supposed to be.

Reply:
I'm not sure what you both find strange about an optional function? 😕
Or why it would not work, given that this was the old behavior. 🤷‍♂

Reply:
there's nothing strange about the optional function (and I've already confirmed that it works, or at least suppresses the warning). I just think it's weird to say "hey, this thing is a function" and then making the default value something that is not a function. It would be like defining something as `PropTypes.bool` and then setting the default to an empty string or something.

Reply:
we could also just _not_ provide a default prop? :woman_shrugging:

Reply:
>I just think it's weird to say "hey, this thing is a function" and then making the default value something that is not a function.

We're not doing that. We're saying "Hey, this thing is an _optional_ function", then omitting it by default.

Using a "no-op" function as a fallback is a pattern I try to avoid, because it can cover up legitimate errors. For example, if `backButtonAction` had been called and we set `showBackButton` to `true` without setting an action, then we'd _want_ it to blow up. That would be a good thing, because we don't want non-functioning back buttons.

In this specific instance it doesn't _really_ matter. Setting the no-op function would have some... infinitesimal performance cost, and that's about it. But I think I have good reason to dislike this pattern in general, and I'd hate to see us make a habit of doing this for _all_ optional functions.

Whether to set `undefined` when we _don't_ want a default is something you'll have to take up with @whymarrh  :sweat_smile: . I can see the case for it if we're planning to use the eslint rule to ensure we don't forget default props, as it's the only reasonable way to _explicitly_ delcare that you're intentionally omitting a default.

Reply:
I think having a default prop is good. I'm indifferent about what it is. To me, setting the `propType` to `PropTypes.func` is a way of saying that this thing is a function, and setting `defaultProp` is an easy way of making it one even when one isn't passed in.

> I'm not sure what you both find strange about an optional function?

It's not strange, it's still an optional function, but now when it's not passed in it's still a function. This, in theory, simplifies our rendering logic to not have to handle `undefined`/`null`. Yes, we're not currently using this particular prop in a way that we'd need to handle either (i.e. we only conditionally use it and we're passing it directly to another component) but further uses of this prop can now rest assured it's always a function, as its type implies.

> Or why it would not work, given that this was the old behavior. 

It would work (as Ricky also noted) but the old behaviour was pre-`propTypes`, so while it worked and would continue to work with our usage, it doesn't feel right now that we've got `propTypes` set.

Reply:
> and I'd hate to see us make a habit of doing this for all optional functions

okay, I've come around. I'm going to make it `undefined`

Reply:
Sorry I didn't mean to make a big deal out of this 😅.

At best the pattern is a useful shorthand alternative to `if (func) { func() }`, and at worse it's a great example of why dynamic types can be painful to deal with. I don't have any opposition to the former.

This case is squarely in the middle in that it's neither good nor bad, it doesn't matter at all, and I only brought this up because I was confused by the decision and reminded of how I was burned in the past by no-op functions. I have wasted much time, my apologies. I'll save this argument for a case when it matters.

Reply:
I think this is fine, this is the correct level of review we should give every PR. We just have to make sure we do this on every PR.